resources/lib/functions.py

```diff
@@ -585,12 +585,16 @@ def processDirectory(results, progress, params):
     name_format = params.get("name_format", None)
     if name_format is not None:
         name_format = urllib.unquote(name_format)
-        name_format = settings.getSetting(name_format)
-
+        name_formatting = settings.getSetting(name_format)
+        if name_format == "latest_name_format":
+            name_formatting = settings.getSetting("episode_name_format")
     dirItems = []
-    result = results.get("Items")
-    if result is None:
+    if results is None:
         result = []
+    if "Items" in results:
+        result = results.get("Items") 
```

Comment:
see my comment below. I think we are very close to having what we need to apply this at a type level already.